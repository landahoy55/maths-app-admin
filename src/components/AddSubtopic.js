import React, { Component } from 'react';
import AddSubTopicForm from './AddSubTopicForm';
import AddSubTopicStyles from './AddSubTopic.css';

class AddSubtopic extends Component {
    
    constructor(props) {
        super(props);

        this.state = {
            addSub: false,
            sudAdded: false,
            title: '',
            description: ''
        }

        //bind function
        this.onNotificationSubmit = this.onNotificationSubmit.bind(this);
        this.addSub = this.addSub.bind(this)
        this.closeSub = this.closeSub.bind(this)
    }

    onNotificationSubmit(event){
         
    }

    addSub(){
        this.setState({
            addSub: true
        })   
    }

    closeSub(sub){

        console.log("Data sent back", sub)

        this.setState({
            addSub:false,
            subAdded: true,
            title: sub.data.title,
            description: sub.data.description
        })
    }

    render() {
    
      return (
        <div>
            <div className="list-group">
                <div className="list-group-item">
                    
                    {
                        this.state.subAdded ? (

                            <div className="list-group">
                            <div className="list-group-item">
                                <div class="alert alert-success" role="alert">
                                    <strong>Sub-topic {this.props.stage} Created</strong>
                                </div>
                                    <h6>Name: {this.state.title}</h6>
                                    <h6>Description: {this.state.description}</h6>
                                </div>
                            </div>
                        ) 
                        : (
                            <div>

                            {
                                this.state.addSub ? (
                                   
                                   <AddSubTopicForm stage={this.props.stage} parentid={this.props.parentid} closeSub={this.closeSub}/>
                                ):
                                (
                                    <div>
                                        <h5>Sub-topic {this.props.stage}</h5>
                                        <button onClick={this.addSub} className="btn btn-primary addBtn">Add</button>
                                    </div>
                                )
                            }

                            </div>
                        )
                    }
                </div>
            </div>
        </div>
      );
    }
  }
  
  
  
  export default AddSubtopic;