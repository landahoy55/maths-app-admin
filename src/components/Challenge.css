.Heading {
    padding-top: 50px;
}

.ChallengeLink {
    color: purple;
    text-decoration: none;
}

.ChallengeLink :hover {
    color: grey;
    text-decoration: none;
}