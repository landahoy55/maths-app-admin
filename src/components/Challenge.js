import React, { Component } from 'react';
import axios from 'axios';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

class Challenge extends Component {
    
    constructor(props) {
        super(props);

        // //bind function
        // this.onNotificationSubmit = this.onNotificationSubmit.bind(this);

        // this.state = {
        //     notificationMessage: '',
        //     notificationConfirmation: ''
        // }
    }

    render() {
    
      return (
        <div className="Challenge App">
            <h1>Daily Challenge</h1>
            <Link to="/todaychallenge"><h2>Create Today's Challenge</h2></Link>
        </div>
      );
    }
  }
  
  
  
  export default Challenge;