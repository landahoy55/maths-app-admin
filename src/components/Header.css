.bg-light {
    /* background: #442DB3 !important; */
    background: linear-gradient(to right,#442DB3, #C86DD7);
}

.navbar-light a {
    color: white !important;
}

.navbar-light a:hover {
    color:rgb(179, 177, 179) !important;
}