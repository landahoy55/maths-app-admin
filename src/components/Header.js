import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link, withRouter } from "react-router-dom";
import AuthenticationTracking from '../AuthenticationTracking';
import axios from 'axios';
import HeaderStyles from './Header.css'
    
class Header extends Component {
    
    constructor(props) {
        super(props);

        this.state = {
            stat: false
        };

        this.removeToken = this.removeToken.bind(this);
    }

    componentWillMount() {
        //token storage
    const token = localStorage.getItem('token')

    //check token...
    if (token) {
        let authTracking = AuthenticationTracking.getInstance();
        let currentComponent = this;
        //authenticate user - need to add serverside auth checker
        console.log("I am here...");
        console.log(token);
            //check token, update auth status.
            axios.get('https://morning-journey-26383.herokuapp.com/v1/account/me', 
            { 
                headers: { Authorization: "Bearer " + token 
            }})
            .then(function (response) {
                console.log("Checking auth... all good")
                console.log(response);
                //log in
                authTracking.setAuthStatus(true);

                currentComponent.setState({
                    stat: true
                })
            })
            .catch(function (error) {
                console.log("Issue with token... all good")
                console.log(error);
                //don't log in
                authTracking.setAuthStatus(false);

                currentComponent.setState({
                    stat: false
                })
            });
        }
    }

    removeToken() {
        //remove token
        console.log("TOKEN REMOVED")
        localStorage.removeItem('token')
        //go to to login
        let currentComponent = this;
        currentComponent.props.history.push("/SignIn");

    }

    render() {
    
    // const authTracker = AuthenticationTracking.getInstance();
    // const status = authTracker.getAuthStatus();
    // console.log("THIS IS THE STATUS", status);
    
    // <ul className="navbar-nav mr-auto">
    //     <li className="nav-item">
    //         <Link className="nav-link" to="/notification">Send Notification</Link>
    //     </li>
    //     <li className="nav-item">
    //         <Link className="nav-link" to="/addchallenge">Add Challenge</Link>
    //     </li>
    //     <li className="nav-item">
    //         <Link className="nav-link" to="/mananagequestions">Manage Questions</Link>
    //     </li>
    //     <li className="nav-item">
    //         <button onClick={this.removeToken}>Log Out</button>
    //     </li>
    // </ul>

    // navbar-light bg-light
    // <a className="nav-link" onClick={this.removeToken}>Log Out</a>
    // <Link className="nav-link" to="/signin">Sign In</Link>

    return (

        <div>

        {
            this.state.stat ? (
                <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <Link className="navbar-brand" to="/">Home</Link>
                    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                        <span className="navbar-toggler-icon"></span>
                    </button>
                    <div className="collapse navbar-collapse" id="navbarSupportedContent">
                        <ul className="navbar-nav mr-auto">   
                        <li className="nav-item">
                            
                        </li>
                        </ul>
                    </div>
                </nav>
            ) : (
                <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <Link className="navbar-brand" to="/">Home</Link>
                    <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                        <span className="navbar-toggler-icon"></span>
                    </button>
                    <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav mr-auto">
                        <li className="nav-item">
                        
                        </li>
                    </ul>
                    </div>
                </nav>
            )
        }
            

        </div>
        

      );
    }
  }
  
  
  
  export default withRouter(Header);


  