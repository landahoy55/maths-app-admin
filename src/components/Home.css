.image-log-in {
    float: left;
}

/* .title-log-in {
    float: right;
} */

.words-log-in {
    top: 30px;
}

.page-header {
    margin-bottom: 50px;
}

.card-body {
    height: 142px;
}

.homeBtn {
    background: #442DB3 !important;
    border-color: #442DB3 !important;
}
    