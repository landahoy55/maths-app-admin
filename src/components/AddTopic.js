import React, { Component } from 'react';
import axios from 'axios';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Subtopic from './Subtopic';
import AddSubtopic from './AddSubtopic';
import AddTopicFormStyles from './AddTopic.css';


class AddTopic extends Component {
    
    constructor(props) {
        super(props);
        
        this.state = {
            isEdit: false,
            created: false,
            title: '',
            description: '',
            createdid: '',
            errorMessage: ''
        }

        this.onEdit = this.onEdit.bind(this);
        this.onEditSubmit = this.onEditSubmit.bind(this);
        this.onSave = this.onSave.bind(this);

    }

    componentDidMount() {
       
    }

    onEdit() {
        this.setState({ isEdit: true });
    }

    onSave(){
        console.log('Testing');
    }

    onEditSubmit(event) {
        event.preventDefault();
        let currentComponent = this

        console.log("Here")

        let title = this.titleInput.value
        let description = this.descriptionInput.value

        if (title == '' || description == '') {

            console.log("In if block")
            
            this.setState({
                errorMessage: "Please add a name and description"
            });

        } else {

            //networking then move to add subtopic screen.
        //submit to server - replace not needed. AXIOS handles JSON parsing. Object was being parsed twice...
        axios.post('https://morning-journey-26383.herokuapp.com/v1/setup/createtopic', {
            title: title,
            description: description
        })
        .then(function (response) {
          console.log("Topic added");
          console.log(response);

          //set state
            currentComponent.setState({
                created: true,
                title: response.data.newTopic.title,
                description: response.data.newTopic.description,
                createdid: response.data.newTopic._id
            })


        })
        .catch(function (error) {
          console.log("Issue saving topic");
          console.log(error);
          alert("issue saving topic")
        });

        }

    }

    render() {
    
      return (
        <div className="Wrapper">

            <div className="page-header">
                <h1>Add Topic</h1>
                <h6>Include a topic name and description</h6>
            </div>

            {
                this.state.created ? (
                    <div>
                        <div className="list-group">
                            <div className="list-group-item">
                                <div className="alert alert-success" role="alert">
                                    <strong>Topic Created</strong> Now to add some questions...
                                </div>
                                <h6><b>Topic Name: </b>{this.state.title}</h6>
                                <h6><b>Topic Description: </b> {this.state.description}</h6>
                                
                            </div>
                        </div>

                    
                       <AddSubtopic stage={1} parentid={this.state.createdid} />
                       <AddSubtopic stage={2} parentid={this.state.createdid} />
                       <AddSubtopic stage={3} parentid={this.state.createdid} />
                       <AddSubtopic stage={4} parentid={this.state.createdid} />
                       <AddSubtopic stage={5} parentid={this.state.createdid} />

                    </div>
                ) : (
                    <div className="list-group">
                        <div className="list-group-item">
                            <form onSubmit={this.onEditSubmit}>
                                <div className="form-group row">
                                    <label htmlFor="titleInput" className="col-sm-3 col-form-label"><b>Topic Name:</b></label>
                                    <div className="col-sm-9">
                                        <input className="form-control" placeholder="Title" ref={titleInput => this.titleInput = titleInput} defaultValue={this.state.title}/>
                                    </div>
                                </div>
                                <div className="form-group row">
                                    <label htmlFor="titleInput" className="col-sm-3 col-form-label"><b>Topic Description:</b></label>
                                    <div className="col-sm-9">
                                        <input className="form-control" placeholder="Description" ref={descriptionInput => this.descriptionInput = descriptionInput} defaultValue={this.state.description} />
                                        <p className="text-danger">{this.state.errorMessage}</p>
                                    </div>
                                </div>
                                
                                <button className="btn btn-success">Save</button>
                        </form> 
                        </div>
                    </div>
                )
            }

        </div>
      )
  }
} 
  
export default AddTopic;
