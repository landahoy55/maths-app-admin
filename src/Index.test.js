import React from 'react';
import ReactDOM from 'react-dom';
import ReactShallowRenderer from 'react-test-renderer/shallow';


import Header from './components/Header';
import AddSubtopic from './components/AddSubtopic';
import AddSubTopicForm from './components/AddSubTopicForm';
import AddTopic from './components/AddTopic';
import Challenge from './components/Challenge';
import HandwritingQuestion from './components/HandwritingQuestions';
import Home from './components/Home';
import InputQuestion from './components/InputQuestion';
import ManageQuestions from './components/ManageQuestions';
import MultipleChoiceImages from './components/MultipleChoiceImages';
import MultipleChoiceQuestion from './components/MultipleChoiceQuestion';
import Notification from './components/Notification';
import PrivateRoute from './components/PrivateRoute';
import Question from './components/Question';

import SendNotification from './components/SendNotification';
import SignIn from './components/SignIn';
import Subtopic from './components/Subtopic';
import SubtopicDetail from './components/SubtopicDetail';
import TodayChallenge from './components/TodayChallenge';
import Topic from './components/Topic';
import TopicDetail from './components/TopicDetail';
import VoiceInputQuestion from './components/VoiceInputQuestions';



//Snapshot tests - will take a snapshot of rendered component and highlight when changes have been made
//Press u in the command line to approve changes

test('it should render header', () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render add sub topic', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AddSubtopic />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render add sub topic form', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AddSubTopicForm />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render add topic', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<AddTopic />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render Challenge', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Challenge />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render HandwritingQuestion', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<HandwritingQuestion />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render Home', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Home />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render InputQuestion', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<InputQuestion />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render InputQuestion', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<InputQuestion />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render ManageQuestions', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ManageQuestions />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render multichoice images', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<MultipleChoiceImages />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render multichoice questions', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<MultipleChoiceQuestion/>);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render notif', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Notification />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render private routes', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<PrivateRoute />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});


test('it should render sendnotif', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<SendNotification />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('it should render SignIn', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<SignIn />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});



test('it should render TodayChallenge', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<TodayChallenge />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});



test('it should render VoiceInputQuestionicDetail', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<VoiceInputQuestion />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});



