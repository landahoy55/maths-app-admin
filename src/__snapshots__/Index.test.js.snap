// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render Challenge 1`] = `
<div
  className="Challenge App"
>
  <h1>
    Daily Challenge
  </h1>
  <Link
    replace={false}
    to="/todaychallenge"
  >
    <h2>
      Create Today's Challenge
    </h2>
  </Link>
</div>
`;

exports[`it should render HandwritingQuestion 1`] = `
<div
  className="list-group-item"
>
  <label
    class="col-form-label"
    for="challengeType"
  >
    Question Number: 
  </label>
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Question"
  />
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Answer"
  />
  <div
    className="text-danger"
  />
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`it should render Home 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`it should render InputQuestion 1`] = `
<div
  className="list-group-item"
>
  <label
    class="col-form-label"
    for="challengeType"
  >
    Question Number: 
  </label>
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Question"
  />
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Answer"
  />
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`it should render InputQuestion 2`] = `
<div
  className="list-group-item"
>
  <label
    class="col-form-label"
    for="challengeType"
  >
    Question Number: 
  </label>
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Question"
  />
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Answer"
  />
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`it should render ManageQuestions 1`] = `
<div
  className="Challenge Wrapper"
>
  <div
    className="page-header"
  >
    <h3>
      Manage Questions
    </h3>
    <h6>
      From here you can add, delete, and amend topics and questions
    </h6>
  </div>
  <div
    className="list-group addTopicHeaderParent"
  >
    <div
      className="list-group-item addTopicHeader"
    >
      <div>
        <h3
          className="list-group-item-text"
        >
          Add a new topic and related questions
        </h3>
        <Link
          className="btn btn-primary btnAdd"
          replace={false}
          to="/addtopic"
        >
          Add Topic
        </Link>
      </div>
    </div>
  </div>
  <div
    className="page-header editTopic"
  >
    <h3>
      Edit an existing topic
    </h3>
  </div>
  <p>
    Loading
  </p>
</div>
`;

exports[`it should render SignIn 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`it should render TodayChallenge 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`it should render VoiceInputQuestionicDetail 1`] = `
<div
  className="list-group-item"
>
  <label
    class="col-form-label"
    for="challengeType"
  >
    Question Number: 
  </label>
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Question"
  />
  <input
    class="form-control"
    onBlur={[Function]}
    placeholder="Answer"
  />
  <div
    className="text-danger"
  />
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`it should render add sub topic 1`] = `
<div>
  <div
    className="list-group"
  >
    <div
      className="list-group-item"
    >
      <div>
        <div>
          <h5>
            Sub-topic 
          </h5>
          <button
            className="btn btn-primary addBtn"
            onClick={[Function]}
          >
            Add
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it should render add sub topic form 1`] = `
<div
  className="Wrapper"
>
  <div
    className="list-group"
  >
    <form
      onSubmit={undefined}
    >
      <div
        className="list-group-item"
      >
        <label
          className="col-form-label"
          htmlFor="challengeType"
        >
          <b>
            Sub-topic Type
          </b>
        </label>
        <select
          className="form-control"
          name="challengeType"
          onChange={[Function]}
        >
          <option
            value="multipleChoice"
          >
            Multiple Choice
          </option>
          <option
            value="input"
          >
            Input
          </option>
          <option
            value="multipleChoiceImage"
          >
            Images
          </option>
          <option
            value="voiceInput"
          >
            Voice
          </option>
          <option
            value="handWriting"
          >
            Handwriting
          </option>
        </select>
        <label
          className="col-form-label"
          htmlFor="subDesc"
        >
          <b>
            Title
          </b>
        </label>
        <input
          className="form-control"
          onBlur={[Function]}
          placeholder="Title"
        />
        <span
          className="form-text text-danger"
        />
        <br />
        <label
          className="col-form-label"
          htmlFor="subDesc"
        >
          <b>
            Description
          </b>
        </label>
        <input
          className="form-control"
          onBlur={[Function]}
          placeholder="Description"
        />
        <span
          className="form-text text-danger"
        />
      </div>
      <div
        className="list-group"
      >
        <MultipleChoiceQuestion
          correctAnswer1Selected=""
          correctAnswer2Selected=""
          correctAnswer3Selected=""
          correctAnswer4Selected=""
          correctAnswer5Selected=""
          onAnswerSet={[Function]}
          onCorrectAnswerSet={[Function]}
          onQuestionSet={[Function]}
          question1AnswersError=""
          question1Error=""
          question2AnswersError=""
          question2Error=""
          question3AnswersError=""
          question3Error=""
          question4AnswersError=""
          question4Error=""
          question5AnswersError=""
          question5Error=""
          questionNumber={1}
        />
        <MultipleChoiceQuestion
          correctAnswer1Selected=""
          correctAnswer2Selected=""
          correctAnswer3Selected=""
          correctAnswer4Selected=""
          correctAnswer5Selected=""
          onAnswerSet={[Function]}
          onCorrectAnswerSet={[Function]}
          onQuestionSet={[Function]}
          question1AnswersError=""
          question1Error=""
          question2AnswersError=""
          question2Error=""
          question3AnswersError=""
          question3Error=""
          question4AnswersError=""
          question4Error=""
          question5AnswersError=""
          question5Error=""
          questionNumber={2}
        />
        <MultipleChoiceQuestion
          correctAnswer1Selected=""
          correctAnswer2Selected=""
          correctAnswer3Selected=""
          correctAnswer4Selected=""
          correctAnswer5Selected=""
          onAnswerSet={[Function]}
          onCorrectAnswerSet={[Function]}
          onQuestionSet={[Function]}
          question1AnswersError=""
          question1Error=""
          question2AnswersError=""
          question2Error=""
          question3AnswersError=""
          question3Error=""
          question4AnswersError=""
          question4Error=""
          question5AnswersError=""
          question5Error=""
          questionNumber={3}
        />
        <MultipleChoiceQuestion
          correctAnswer1Selected=""
          correctAnswer2Selected=""
          correctAnswer3Selected=""
          correctAnswer4Selected=""
          correctAnswer5Selected=""
          onAnswerSet={[Function]}
          onCorrectAnswerSet={[Function]}
          onQuestionSet={[Function]}
          question1AnswersError=""
          question1Error=""
          question2AnswersError=""
          question2Error=""
          question3AnswersError=""
          question3Error=""
          question4AnswersError=""
          question4Error=""
          question5AnswersError=""
          question5Error=""
          questionNumber={4}
        />
        <MultipleChoiceQuestion
          correctAnswer1Selected=""
          correctAnswer2Selected=""
          correctAnswer3Selected=""
          correctAnswer4Selected=""
          correctAnswer5Selected=""
          onAnswerSet={[Function]}
          onCorrectAnswerSet={[Function]}
          onQuestionSet={[Function]}
          question1AnswersError=""
          question1Error=""
          question2AnswersError=""
          question2Error=""
          question3AnswersError=""
          question3Error=""
          question4AnswersError=""
          question4Error=""
          question5AnswersError=""
          question5Error=""
          questionNumber={5}
        />
      </div>
      <button
        className="btn btn-success"
        onClick={[Function]}
      >
        Save Subtopic
      </button>
    </form>
    <p />
  </div>
</div>
`;

exports[`it should render add topic 1`] = `
<div
  className="Wrapper"
>
  <div
    className="page-header"
  >
    <h1>
      Add Topic
    </h1>
    <h6>
      Include a topic name and description
    </h6>
  </div>
  <div
    className="list-group"
  >
    <div
      className="list-group-item"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <label
            className="col-sm-3 col-form-label"
            htmlFor="titleInput"
          >
            <b>
              Topic Name:
            </b>
          </label>
          <div
            className="col-sm-9"
          >
            <input
              className="form-control"
              defaultValue=""
              placeholder="Title"
            />
          </div>
        </div>
        <div
          className="form-group row"
        >
          <label
            className="col-sm-3 col-form-label"
            htmlFor="titleInput"
          >
            <b>
              Topic Description:
            </b>
          </label>
          <div
            className="col-sm-9"
          >
            <input
              className="form-control"
              defaultValue=""
              placeholder="Description"
            />
            <p
              className="text-danger"
            />
          </div>
        </div>
        <button
          className="btn btn-success"
        >
          Save
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`it should render header 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`it should render multichoice images 1`] = `
<div
  className="list-group-item"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <label
        className="col-form-label"
        htmlFor="challengeType"
      >
        Question: 
      </label>
    </div>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Enter a question..."
      />
      <span />
      <span />
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="radio"
      >
        <label
          className="radio-inline"
        >
          <input
            name={undefined}
            onChange={[Function]}
            type="radio"
            value="answer1"
          />
           Correct answer
        </label>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer2"
      />
       Correct answer
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer3"
      />
       Correct answer
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer4"
      />
       Correct answer
      <br />
    </div>
  </div>
  <span />
  <span />
  <span />
  <span />
  <span />
  <div>
    <t
      accept="image/*"
      disableClick={false}
      disablePreview={false}
      disabled={false}
      maxSize={Infinity}
      minSize={0}
      multiple={false}
      onDrop={[Function]}
      preventDropOnDocument={true}
      style={
        Object {
          "border": "2px dashed #442DB3",
          "height": "20%",
          "width": "20%",
        }
      }
    >
      <p>
        Drag in an image, or click to upload
      </p>
    </t>
    <p
      className="text-danger"
    />
  </div>
</div>
`;

exports[`it should render multichoice questions 1`] = `
<div
  className="list-group-item"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2"
    >
      <label
        className="col-form-label"
        htmlFor="challengeType"
      >
        Question: 
      </label>
    </div>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Enter a question..."
      />
      <span />
      <span />
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <div
        className="radio"
      >
        <label
          className="radio-inline"
        >
          <input
            name={undefined}
            onChange={[Function]}
            type="radio"
            value="answer1"
          />
           Correct answer
        </label>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer2"
      />
       Correct answer
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer3"
      />
       Correct answer
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-6"
    >
      <input
        className="form-control"
        onBlur={[Function]}
        placeholder="Possible answer..."
      />
    </div>
    <div
      className="col-sm-6"
    >
      <input
        name={undefined}
        onChange={[Function]}
        type="radio"
        value="answer4"
      />
       Correct answer
      <br />
    </div>
  </div>
  <span />
  <span />
  <span />
  <span />
  <span />
</div>
`;

exports[`it should render notif 1`] = `
<div
  className="Wrapper"
>
  <div
    className="SendNotification"
  >
    <div
      className="page-header"
    >
      <h3>
        Push Notifications
      </h3>
      <h6>
        Send a short message to all users that have given permission to receive notifications
      </h6>
    </div>
    <SendNotification
      buttonText="Send"
      onNotificationSubmit={[Function]}
    />
    <div>
      <div
        className="alert"
        role="alert"
      >
        <h4
          className="alert-heading"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`it should render private routes 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`it should render sendnotif 1`] = `
<div
  className="SendNotification"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        placeholder="Notificaiton"
      />
      <small
        className="form-text text-muted"
        id="emailHelp"
      >
        Try to keep the notification short and engaging
      </small>
      <button
        className="btn btn-success notificationBtn"
      />
    </div>
  </form>
</div>
`;
